tsdb/engine/tsm1/writer.go

```diff
@@ -264,9 +264,10 @@ func NewDirectIndex() TSMIndex {
 // directIndex is a simple in-memory index implementation for a TSM file.  The full index
 // must fit in memory.
 type directIndex struct {
-	mu     sync.RWMutex
-	size   uint32
-	blocks map[string]*indexEntries
+	mu               sync.RWMutex
+	size             uint32
+	blocks           map[string]*indexEntries
+	cachedSortedKeys []string
```

Comment:
This will double the size of the index in memory when writing TSM files during compactions.  Can you revert this?